### ISSUE TITLE
pauseAsync has to be called multiple times in order for a consecutive playAsync to play the sound

### ISSUE THREAD

**Issue by Adherentman** [expo/expo#1141]
```javascript
async _playRecording() {
    const { getUri, recordingDuration } = this.props;
    const { sound, status } = await Audio.Sound.create(
      { uri: getUri } as any,
      {
        shouldPlay: true,
        isLooping: true,
      },
      this._updateScreenForSoundStatus,
    );
    this.sound = sound;
    this.setState({
      playingStatus: 'playing'
    });
  };
```
```javascript
async _pauseAndPlayRecording() {
    if (this.sound != null) {
      if (this.state.playingStatus == 'playing') {
        await this.sound.pauseAsync();
        this.setState({
          playingStatus: 'donepause',
        });
      } else {
        await this.sound.playAsync();
        this.setState({
          playingStatus: 'playing',
        });
      }
    }
  }

```
```javascript
  _playAndPause = () => {
    switch (this.state.playingStatus) {
      case 'nosound':
        this._playRecording();
        break;
      case 'donepause':
      case 'playing':
        this._pauseAndPlayRecording();
        break;
    }
  }

```

I recorded the audio, click play, the audio plays normally but I click on the pause and click again to play, playAsync () is not work!!!  **I must pauseAsync() twice in ios 11.2.1, The playAsync() can work!**

**Comment by sjchmiela** [comment]
Hi, @Adherentman!

I've created a snack based on your code that shows how you could make better use of the `Sound` API.

https://snack.expo.io/SkjbMrFGM

Let me know if it helps you resolve issues with your application. 🙂

**Comment by anp** [comment]
Hi! Thanks @sjchmiela for that :D. @Adherentman if this doesn't resolve your problem please post back and we can take another look, I'll close this for now.

**Comment by Adherentman** [comment]
@sjchmiela @anp Hi,Thanks Help.

Yesterday I tried on my project and found that I had to execute `pasueAsync()` twice, then `playAsync()` will working!😥  

This problem in @sjchmiela create project too.  

But this problem is not on Android. I think this is the ios(11.2.1) problem🙁

**Comment by sjchmiela** [comment]
I cannot reproduce this issue. ☹️

**What do you mean by “`playAsync` doesn't work”? Do you mean it never resolves or it never plays the sound or it throws an error?**

On Appetize it seems to work properly (I cannot hear the sound, but the promises resolve properly, as you can see).

![dec-22-2017 10-05-18](https://user-images.githubusercontent.com/1151041/34292110-ae6842cc-e6ff-11e7-85ff-61dcb0c66bd0.gif)

I also tested the code against both the latest `master` version of Expo and SDK 24.0.2 and everything worked properly.

**Comment by Adherentman** [comment]
@sjchmiela  OK, 
First I recorded the sound, and then I clicked the play button, after which I paused, I needed to pause many times and click play again to play my recorded audio again。

**Comment by sjchmiela** [comment]
Ok, let me rephrase that:

1. Recorded sound, playing the sound back,
2. Pause,
3. Resuming the sound doesn't work,
4. Pause, pause, pause,
5. Resuming the sound works.

Is that correct?

**Comment by Adherentman** [comment]
@sjchmiela Yes,Yes,Yes😳

**Comment by Adherentman** [comment]
Hi, are you still there @sjchmiela

**Comment by sjchmiela** [comment]
How do you know when the initial `pauseAsync` resolves? Let's look at this situation:

```
+----+---->  The sound is playing
     |
     |
     +---->  pauseAsync  +  pauseAsync starts
     |                   |
     |                   |
     +---->  playAsync   |
     |                   |
     |                   |
     +---->  pauseAsync  |
     |                   |
     |                   v  pauseAsync resolves
     |                      The sound paused
     |
     +---->  playAsync   +  playAsync starts
                         |
                         |
                         v  The sound is playing
```

Here it seems that multiple calls to `pauseAsync` stopped the sound, however, it really only happened because of the first `pauseAsync`. So you have to ensure that all the promises are awaited properly.

Maybe somewhere else in the code another promise isn't awaited and the first `pauseAsync` is somehow lost?

I can't reproduce your issue. :( I used https://snack.expo.io/BkX2ZNfmG to record this video: https://youtu.be/A2LNkaixw44. The sound is properly paused and played and the UI updates accordingly. Maybe you could provide me with a Snack and a video that would demonstrate this bug?

**Comment by Adherentman** [comment]
@sjchmiela, 
When I want to demonstrate this bug,But I after one of the two `pauseAsync` is deleted, it works. 
I promise to have this bug before mentioning Issue!
Sorry, I do not know what I did.

**Comment by sjchmiela** [comment]
I'm happy for you! :) I'll close this for now, feel free to reopen this once it becomes reproducible again.

**Comment by Adherentman** [comment]
@sjchmiela Thank you soo much!
